### PR TITLE
Remove contract name for add and update contract commands

### LINKED ISSUE #231: Deploying contract without name
Could we remove the name parameter when deploying a contract or at least make it optional since it must match the name of the contract from file which could just be read from the file itself.

### PR DESCRIPTION
**Pull request by sukantoraymond**
Closes #231 

This PR enables users to add and update contract without using contract name. It will extract contract name from contract source code.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by sukantoraymond** [comment]
I don't think its necessary to make contract name optional in the add / update contract command? We can just remove it altogether @sideninja

**Comment by sideninja** [comment]
@sukantoraymond we need to keep it due to backward compatibility.

File: internal/accounts/contract-add.go [hunk 1]
```diff
@@ -57,14 +59,34 @@ func addContract(
...
+	} else {
+		filename = args[1]
+	}
 
 	code, err := readerWriter.ReadFile(filename)
 	if err != nil {
 		return nil, fmt.Errorf("error loading contract file: %w", err)
 	}
 
+	resolver, err := contracts.NewResolver(code)
```
Comment (sideninja) [inline internal/accounts/contract-add.go:76]: This should be done inside the account service.  Resolver is already created there https://github.com/onflow/flow-cli/blob/master/pkg/flowkit/services/accounts.go#L309

File: pkg/flowkit/contracts/resolver.go [hunk 2]
```diff
@@ -121,3 +121,23 @@ func (r *Resolver) getFileImports() []string {
 
 	return imports
 }
+
+func (r *Resolver) GetSourceContractName() (string, error) {
```
Comment (sideninja) [inline pkg/flowkit/contracts/resolver.go:125]:
```suggestion
func (r *Resolver) Name() (string, error) {
```

File: docs/account-add-contract.md [hunk 3]
```diff
@@ -6,13 +6,18 @@ sidebar_title: Deploy a Contract
...
 ```shell
-flow accounts add-contract <name> <filename> [<argument> <argument>...] [flags]
+flow accounts add-contract <filename> [<argument> <argument>...] [flags]
 ````
 
+⚠️  Deprecation notice: using name argument in adding contract command will be deprecated soon.
+```shell
+flow accounts add-contract <name> <filename> [<argument> <argument>...] [flags]
+```
+
```
Comment (sideninja) [inline docs/account-add-contract.md:16]: Deprecations are not needed in documentation

File: pkg/flowkit/contracts/resolver_test.go [hunk 4]
```diff
@@ -141,5 +141,24 @@ func TestResolver(t *testing.T) {
 			assert.Equal(t, cleanCode(code), cleanCode(resolved[i]))
 		}
 	})
+	t.Run("Get Contract Name", func(t *testing.T) {
```
Comment (sideninja) [inline pkg/flowkit/contracts/resolver_test.go:144]: 👍 nice test

File: internal/accounts/contract-update.go [hunk 5]
```diff
@@ -57,8 +57,13 @@ func updateContract(
...
 	state *flowkit.State,
 ) (command.Result, error) {
-	name := args[0]
-	filename := args[1]
+	filename := ""
+
+	if len(args) == 1 {
+		filename = args[0]
+	} else {
+		filename = args[1]
```
Comment (sideninja) [inline internal/accounts/contract-update.go:65]: deprecation notice

File: pkg/flowkit/contracts/resolver.go [hunk 6]
```diff
@@ -121,3 +121,24 @@ func (r *Resolver) getFileImports() []string {
 
 	return imports
 }
+
+// Name returns the name of the provided contract program
+func (r *Resolver) Name() (string, error) {
```
Comment (sideninja) [inline pkg/flowkit/contracts/resolver.go:126]: Perfect!

File: internal/accounts/contract-update.go (hunk 5 above)
Comment (DylanTinianov) [reply internal/accounts/contract-update.go:69]: Added

File: internal/accounts/contract-add.go [hunk 7]
```diff
@@ -57,8 +59,14 @@ func addContract(
 	srv *services.Services,
 	state *flowkit.State,
 ) (command.Result, error) {
-	name := args[0]
-	filename := args[1]
+	filename := ""
+	if len(args) == 1 {
+		filename = args[0]
+	} else {
+		output.NewStdoutLogger(output.InfoLog).Info("⚠️Deprecation notice: using name argument in add contract " +
```
Comment (sideninja) [inline internal/accounts/contract-add.go:66]: just use `fmt.Println` here.

Comment (DylanTinianov) [reply internal/accounts/contract-add.go]: Done

**Pull request merged**